yt/visualization/plot_modifications.py

```diff
@@ -77,8 +77,10 @@ def _project_coords(self, plot, coord):
         """
         if len(coord) == 3:
             if not isinstance(coord, YTArray):
-                coord = plot.data.ds.arr(coord, "code_length")
-            coord.convert_to_units("code_length")
+                coord_copy = plot.data.ds.arr(coord, "code_length")
+            #coord is being copied so that if the user has a unyt_array already
+            #we don't change the user's version
+            coord_copy = coord.to("code_length")
```

Comment:
I think this should fix the bug that was introduced by changing the variable names
```suggestion
                coord_copy = plot.data.ds.arr(coord, "code_length")
            else:
                #coord is being copied so that if the user has a unyt_array already
                #we don't change the user's version
                coord_copy = coord.to("code_length")
```